user.go

```diff
@@ -39,6 +42,58 @@ func (a *App) GetUser(userID string) (*User, error) {
 	}
 }
 
+// GetUserByIdentifier gets a user using their identifier
+// returns user on success, error on failure
+func (a *App) GetUserByIdentifier(identifier string) (*User, error) {
+	var errorText string
+	message := "failed to get Passage User By Identifier"
+	limit := 1
+	res, err := a.client.ListPaginatedUsersWithResponse(
+		context.Background(),
+		a.ID,
+		&ListPaginatedUsersParams{
+			Limit:      &limit,
+			Identifier: &identifier,
+		},
+	)
+
+	if err != nil {
+		return nil, Error{Message: "network error:failed to get Passage User by Identifier"}
+	}
+
+	if res.JSON200 != nil {
+		users := res.JSON200.Users
+		if len(users) == 0 {
+			message = fmt.Sprintf(IdentifierDoesNotExist, identifier)
+			return nil, Error{
+				Message:    message,
+				StatusCode: 404,
+				StatusText: "404 not found",
+				ErrorText:  "user_not_found",
```

Comment:
These are hard coded. I tried to use `N404ErrorCode` and the types weren't matching. Taking suggestions on what other generated variables I could use instead of theses.

Reply:
```suggestion
				StatusCode: http.StatusNotFound,
				StatusText: http.StatusText(http.StatusNotFound),
				ErrorText:  string(UserNotFound),
```

Reply:
made these changes